### DISCUSSION TITLE
Register default handler for blazor webview with TryAddHandler?

### DISCUSSION THREAD

**Discussion by miri10134** [dotnet/maui#6222] [Ideas]
Hello. After migrating to MAUI RC1 I found my custom BlazorWebViewHandler not working anymore, because the new AddMauiBlazorWebView-Method includes the [following line](https://github.com/dotnet/maui/blob/9f4bb060eceb6e1b05b3a3b8000a501a7f2bffbb/src/BlazorWebView/src/SharedSource/BlazorWebViewServiceCollectionExtensions.cs#L41):

```csharp
services.ConfigureMauiHandlers(static handlers => handlers.AddHandler<IBlazorWebView, BlazorWebViewHandler>());
```

This requires custom implementations for IBlazorWebView-Handlers to be added after calling the method. 
For example, the following code, which is just the default template with a custom handler configuration, will not work as intended:

```csharp
var builder = MauiApp.CreateBuilder();
builder
	.UseMauiApp<App>()
	.ConfigureFonts(fonts =>
	{
		fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
	})
	.ConfigureMauiHandlers(handlers => handlers.AddHandler<IBlazorWebView, MyBlazorWebViewHandler>());

builder.Services.AddMauiBlazorWebView();

return builder.Build();
```

Could the handler registration in BlazorWebViewServiceCollectionExtensions be changed to .TryAddHandler to prefer custom handlers, so developers don't need to be aware of the registration order? 

```csharp
services.ConfigureMauiHandlers(static handlers => handlers.TryAddHandler<IBlazorWebView, BlazorWebViewHandler>());
```

**Comment by richardhauer** [comment]
Thanks for logging this - I was trying to work out why my handlers were not getting connected.
